data-preprocessing/crop_riddle_script/crop_riddle_fs_video.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python3
+
+# Import libraries
+import sys # System
+import whisper  # Speech-to-text
+import torch 
+from moviepy.editor import VideoFileClip  # Video clipping
+import os # Working with directory
+import json # Working with JSON
+import re # Regular expression
+import argparse
+
+# Function to display help text # Add argparse to this as well as a main function
+def print_help():
+    """
+    Print usage instructions
+    """
+    print("Usage: python crop_riddle_fs.py [OPTIONS]")
+    print("")
+    print("Options:")
+    print("  -h, --help          Display this help message and exit")
+    print("  -v, --version       Display the script version and exit")
+    print("  -i, --instruction   Provide the path to the instruction file")
+    print("")
+    print("Usage Instructions:")
+    print("- Provide the root path of videos in the instruction file.")
+    print("- Provide the destination path of cropped videos in the instruction file.")
+    print("- If necessary, configure the target statements in the instruction file.")
+    print("- Ensure the videos in the root path end with an mp4 extension.")
+    sys.exit(0)
+
+# -------------------------------------------------
+# Function to read instruction file root directory
+#--------------------------------------------------
+def readInstructionFile():
+    """
+    Read instruction file
+    """
+    with open('ins.json', 'r') as ins:
+        data = json.load(ins)
+    return data
+
+# Function to validate json instruction file
+def check_json_file(filepath):
+    """
+    Check required keys inside json file
+    """
+    required_keys = ["root_path", "cropped_video_destination_path"]
+
+    with open(filepath) as f:
+        data = json.load(f)
+
+    for key in required_keys:
+        if key not in data:
+            raise ValueError("Check required parameters in instruction file")
+
+    return True
+
+# Check json file
+print("Checking instruction file...")
+check_json_file("ins.json")
+
+# Read instructions from file
+data = readInstructionFile() 
+    
+# Get root path and destination path from instructions
+root_path = data["root_path"] 
+
+# Check if root path is absolute
+print("Checking if root path provided is an absolute path...")
```

Comment:
why does the root path need to be absolute?

Reply:
The root path can be both absolute or relative. The team decided to make root path absolute to ensure consistent use among other users. Also, the videos to be cropped may not be in a relative path of the script. Thus, users providing absolute paths will minimise issues with relative vs. absolute.